src/ffi_baggage.h

```diff
@@ -0,0 +1,9 @@
+#include "fish_indent_common.h"
+
+// Symbols that get autocxx bindings but are not used in a given binary, will cause "undefined
+// reference" when trying to link that binary. Work around this by marking them as used in
+// all binaries.
+void mark_as_used() {
+    //
+    pretty_printer_t({}, {});
+}
```

Comment:
I think `__attribute__ ((used))` on the constructor [should accomplish the same](https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#Common-Function-Attributes:~:text=for%20this%20function.-,used,-This%20attribute%2C%20attached)? 

> This attribute, attached to a function, means that code must be emitted for the function even if it appears that the function is not referenced. This is useful, for example, when the function is referenced only in inline assembly.
> 
> When applied to a member function of a C++ class template, the attribute also means that the function is instantiated if the class itself is instantiated.

Reply:
hmm `__attribute__((used))` does not seem to work